### DISCUSSION TITLE
Custom LLM LLAMA model for freecad for object generation

### DISCUSSION THREAD

**Discussion by hemangjoshi37a** [FreeCAD/FreeCAD#18772] [Code contribution discussions]
Please if anyone can sponsor me for this then I want to fine-tune the model specifically for generating and editing freecad parts and doing other things such as simulations. We will use llama 3.3 models for this and sloth for fine-tuning.

Here is my GitHub profile : https://github.com/hemangjoshi37a

**Comment by hemangjoshi37a** [comment]
We will make an add-on from which users can contribute their data to our dataset in which users can describe what they are doing or have done and all the keyboard mouse interactions and screenshots will be saved in the dataset on which we will fine-tune out model . Here is my library for recording computer interactions for large action model fine-tuning : https://github.com/hemangjoshi37a/AIComputerInteractionLogger

**Reply by Bardo-Konrad** [reply]
[related](https://forum.freecad.org/viewtopic.php?t=96135)

**Comment by hemangjoshi37a** [comment]
And then we can have a leadboard for the contributors as webpage or in the GitHub repo.

**Comment by yorikvanhavre** [comment]
You can always request a grant from the FPA - see https://fpa.freecad.org/programs/fpadf-announcement

**Reply by Bardo-Konrad** [reply]
[related](https://forum.freecad.org/viewtopic.php?t=96135)
